README.md

```diff
@@ -140,43 +152,43 @@ features = ["from", "add", "iterator"]
 
 And this to the top of your Rust file:
 ```rust
-extern crate derive_more;
 // use the derives that you want in the file
 use derive_more::{Add, Display, From};
 ```
 If you're still using Rust 2015, add this instead:
-```rust
+```rust,edition2015
 extern crate core;
 #[macro_use]
 extern crate derive_more;
+# fn main() {} // omit wrapping statements above into `main()` in tests
 ```
 
 [`cargo-expand`]: https://github.com/dtolnay/cargo-expand
 [`derive-new`]: https://github.com/nrc/derive-new
 
-[`From`]: https://jeltef.github.io/derive_more/derive_more/from.html
-[`Into`]: https://jeltef.github.io/derive_more/derive_more/into.html
-[`FromStr`]: https://jeltef.github.io/derive_more/derive_more/from_str.html
-[`TryInto`]: https://jeltef.github.io/derive_more/derive_more/try_into.html
-[`IntoIterator`]: https://jeltef.github.io/derive_more/derive_more/into_iterator.html
-[`AsRef`]: https://jeltef.github.io/derive_more/derive_more/as_ref.html
-[`AsMut`]: https://jeltef.github.io/derive_more/derive_more/as_mut.html
-
-[`Display`-like]: https://jeltef.github.io/derive_more/derive_more/display.html
-
-[`Error`]: https://jeltef.github.io/derive_more/derive_more/error.html
-
-[`Index`]: https://jeltef.github.io/derive_more/derive_more/index_op.html
-[`Deref`]: https://jeltef.github.io/derive_more/derive_more/deref.html
-[`Not`-like]: https://jeltef.github.io/derive_more/derive_more/not.html
-[`Add`-like]: https://jeltef.github.io/derive_more/derive_more/add.html
-[`Mul`-like]: https://jeltef.github.io/derive_more/derive_more/mul.html
-[`Sum`-like]: https://jeltef.github.io/derive_more/derive_more/sum.html
-[`IndexMut`]: https://jeltef.github.io/derive_more/derive_more/index_mut.html
-[`DerefMut`]: https://jeltef.github.io/derive_more/derive_more/deref_mut.html
-[`AddAssign`-like]: https://jeltef.github.io/derive_more/derive_more/add_assign.html
-[`MulAssign`-like]: https://jeltef.github.io/derive_more/derive_more/mul_assign.html
-
-[`Constructor`]: https://jeltef.github.io/derive_more/derive_more/constructor.html
-[`IsVariant`]: https://jeltef.github.io/derive_more/derive_more/is_variant.html
-[`Unwrap`]: https://jeltef.github.io/derive_more/derive_more/unwrap.html
+[`From`]: doc/from.md
```

Comment:
These links should point to the docs.rs link instead. Otherwise the readme contains broken links. I think this is another good reason to keep the README and lib.rs docs different (although very similar).

Reply:
@JelteF nope. 

1. These links do work in `README.md` only and are overwritten on the crate level for the generated Rust docs. You can run `cargo doc --open` and see that they point correctly there.

2. As I've pointed out in this PR description, the `doc/*.md` solution is temporary, to not break the README on current `master` after merging this PR, as there are no such docs at `docs.rs` at the moment. Once we release `1.0.0` they can be changed to point to `docs.rs` directly.

Reply:
My main problem is that we want relative path links on docs.rs to make sure the docs main page for old derive_more docs keeps pointing to sub pages for that same version. But the readme should instead use full urls. I don't see any way around this. I also don't like copy pasting docs, but I think it's the least bad option. Another advantage of it is that the readme can stay valid for the previous version of derive_more during development of the next version.

Reply:
The only issue is with the [hidden line exposed on GitHub in example code for 2015 edition](https://github.com/JelteF/derive_more/pull/209#discussion_r996359502).

Reply:
@JelteF ping

Reply:
> After we rollout 1.0.0 release, the links in README on GitHub will point to https://docs.rs/derive_more/latest which is exactly what we need. On docs.rs itself they will remain relative.

Maybe I'm missing something, but that sounds like for every future release we'll have this problem. And we need to do the dance with swapping the links in the readme before we release and after we release.

I think a better way is to have another file than README.md being the source of lib.rs docs. And have a simple `sed` command in a script to transform this to our README. Thus making the README a build artifact of the cannonical docs file.

Something like this:
```bash
sed -e 's=: doc/=: https://docs.rs/derive_more/latest/=g' -e '/# fn main() {}/d' doc/index.md > README.md
```

Reply:
One thing I dislike about the current PR is that it breaks the README until we release.

Reply:
@JelteF as I've mentioned before, the only issue is the `# fn main() {}` line.

> One thing I dislike about the current PR is that it breaks the README until we release.

Why it breaks? It leads to actual `doc/*.md` files. The links work as expected.

If you want them to point to the previous location (`https://jeltef.github.io/derive_more/derive_more/*.html`), then it's OK too. I'll change them until we do the release.

> Maybe I'm missing something, but that sounds like for every future release we'll have this problem.

No. Once new docs hit the `docs.rs`, we change the links in README to `https://docs.rs/derive_more/latest/`, and for any subsequent releases we don't need to do anything.

> And we need to do the dance with swapping the links in the readme before we release and after we release.

No, we don't need to swap anything before a release. We only need to swap links in README after the first `1.0.0` release once, and only because external docs don't exist on `docs.rs` at the moment.

> I think a better way is to have another file than README.md being the source of lib.rs docs. And have a simple bash script with a `sed` command to transform this to our README. Thus making the README a build artifact of the cannonical docs file.

Yes, I've thought about it. But as I've [mentioned before](https://github.com/JelteF/derive_more/pull/209#discussion_r996430435), I think that a single `# fn main() {}` line is not worth this tooling complication.

Reply:
> If you want them to point to the previous location (`https://jeltef.github.io/derive_more/derive_more/*.html`), then it's OK too. I'll change them until we do the release.

Done in https://github.com/JelteF/derive_more/pull/209/commits/c1a11fd5be78f84384b21f48eec3ff26db935957.

Reply:
Thanks for explaining and reverting the links, and yeah I agree now this is a good solution. I didn't realise your link overwriting trick.

For clarification though, I don't think the doc/*.md links would have gone to the github files.

They should have been blob/master/doc/*.md links instead:
1. https://github.com/JelteF/derive_more/blob/master/doc/add.md
2. https://github.com/JelteF/derive_more/doc/add.md